honeybadgermpc/betterpairing.py

```diff
@@ -134,32 +151,33 @@ def __setstate__(self, d):
         self.__init__(d)
 
     def invert(self):
-        negone = PyFr(str(1))
-        negone.negate()
-        self.pyg1.mul_assign(negone)
+        self.pyg1.mul_assign(PyFr(str(1)).negate())
```

Comment:
Maybe premature optimization - but this looks like an easy candidate for caching. 

Separately - It always irks me to see serialization/deserialization to decimal numbers as the way of communicating with an FFI -- we're encountering this in bindings to the C++ NTL library too so I don't have a solution in mind.

Reply:
I see.